mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1227,32 +1257,26 @@ def _get_args_for_metrics(self, extra_metric, eval_df):
                     # case where the param is defined as part of the evaluator_config
                     elif column in self.evaluator_config:
                         eval_fn_args.append(self.evaluator_config.get(column))
+
+                    # case where this is the name of another metric
+                    elif metric_value := self._get_column_in_metrics_values(column):
+                        eval_fn_args.append(metric_value)
+
+                    # in the case that:
+                    # the metric has not been calculated yet, but is scheduled to be calculated
+                    # "before" this metric in self.ordered_metrics, we append None to indicate
+                    # that there is not an error in the dependencies
```

Comment:
how would we end up in such a case? would it be possible to enforce processing in order to avoid this,

Reply:
Since we are using `_get_args_for_metrics()` both when we actually calculate the metric and when we just want to determine dependencies, in the case that we are just determining the dependencies we won't have the value of the metric it is dependent on just yet (since we first want to ensure that there aren't issues with circular dependencies etc before we actually calculate anything). In the case that we want to ensure we have the values before we check args, we would have to calculate metrics before we know if all the dependencies (or columns) are correctly set up/mapped.